XouDevSpec/Classes/Model/ApiCallerModel.swift

```diff
@@ -29,7 +29,7 @@ public struct ApiCallerModel {
     
     public  func getAnimeData(completetionHandler: @escaping(Result<[AnimeDetails], AnimeError>) -> Void) {
     
-        var request = URLRequest(url: NSURL(string:  "https://jikan1.p.rapidapi.com/search/anime?q=\(searchText)")! as URL,
+        var request = URLRequest(url: NSURL(string:"https://jikan1.p.rapidapi.com/search/anime?q=\(searchText)")! as URL,
```

Comment:
With regards to your unwrapping, as requested here's some suggestions:
Firstly, we should try and keep our url strings in a struct or in this case an enum, then you can just pass a searchTextValue for it and use it wherever you need it
Secondly, a simple guard let would work here as is:
```
guard let urlString = "https://jikan1.p.rapidapi.com/search/anime?q=\(searchText)", 
           let url = NSURL(string: urlString) else {
   completionHandler(.failure(.malformedURLError)) //Create a new error type if you like here
}
var request = URLRequest(url: url, cachePolicy.....)
```

Reply:
Also, wouldn't we rather use URL instead of NSURL here ?

Reply:
ok cool thank you. I suck with unwrapping haha.. So its telling me now that it is expecting a optional instead of a string *facepalm*: 
<img width="982" alt="Screenshot 2020-03-27 at 16 40 04" src="https://user-images.githubusercontent.com/60639389/77767473-ccda3f00-7049-11ea-91c9-4d7a5c67ea51.png">
Any suggestions how to fix it and still unwrap it?
PS: I really appreciate the help, I know nothing xD

Reply:
My bad, I went one step further than necessary:
`guard let url = NSURL(string: "https://jikan1.p.rapidapi.com/search/anime?q=\(searchText)") else` is enough, no need to unwrap the string seperately since it's not optional

Reply:
That makes more sense haha. It seems prater is MIA. Would you mind if you have time to help with the weird tableview thing?

Reply:
Haha it is Friday, but with lockdown not much better for me to be doing lol. I'll slack you in a few

Reply:
You can take out the urlString from the guard let and put above it, then you should just have 
` guard let url = URL(...`